qiskit/primitives/base/base_primitive.py

```diff
@@ -72,3 +74,38 @@ def _cross_validate_circuits_parameter_values(
         return validation._cross_validate_circuits_parameter_values(
             circuits, parameter_values=parameter_values
         )
+
+
+BasePrimitive = BasePrimitiveV1
+
+
+class BasePrimitiveV2(ABC):
+    """Primitive abstract base class version 2."""
+
+    version = 2
+    _options_class: type[BasePrimitiveOptions] = BasePrimitiveOptions
+
+    def __init__(self, options: Optional[BasePrimitiveOptionsLike] = None):
+        self._options: type(self._options_class)
+        self._set_options(options)
+
+    @property
+    def options(self) -> BasePrimitiveOptions:
+        """Options for BaseEstimator"""
```

Comment:
```suggestion
        """Options for the primitive"""
```
This doc string will be inherited by subclasses so should leave the type generic